container.go

```diff
@@ -754,3 +775,37 @@ func (c *Container) detachDevices() error {
 
 	return nil
 }
+
+func (c *Container) addResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config == nil {
+		return nil
+	}
+
+	vCPUs := ConstraintsToVCPUs(c.config.Resources.CPUQuota, c.config.Resources.CPUPeriod)
+	if vCPUs != 0 {
+		virtLog.Debugf("hot adding %d vCPUs", vCPUs)
+		if err := c.pod.hypervisor.hotplugAddDevice(uint32(vCPUs), cpuDev); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Container) removeResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config == nil {
+		return nil
+	}
+
+	vCPUs := ConstraintsToVCPUs(c.config.Resources.CPUQuota, c.config.Resources.CPUPeriod)
+	if vCPUs != 0 {
```

Comment:
This cannot be as simple as it is. First, we should check that removing the numbers `vCPUs` here won't fail because we're trying to remove the coldplugged default CPUs (in case we didn't need to hotplug when `addResources` has been called). And second, we should have saved the highest number of CPUs needed by any container on this pod, in order to prevent the removal of CPUs still needed by this container.